lib/libzfs/libzfs_pool.c

```diff
@@ -3852,8 +3852,9 @@ zpool_history_unpack(char *buf, uint64_t bytes_read, uint64_t *leftover,
 		/* add record to nvlist array */
 		(*numrecords)++;
 		if (ISP2(*numrecords + 1)) {
-			*records = realloc(*records,
-			    *numrecords * 2 * sizeof (nvlist_t *));
+			if (!(*records = realloc(*records,
+			    *numrecords * 2 * sizeof (nvlist_t *))))
+				return (ENOMEM);	/* couldn't realloc */
```

Comment:
`nvlist_unpack` above allocates a new nvlist. You need to free it to avoid leaking memory.